api/controllers/courseController.js

```diff
@@ -0,0 +1,119 @@
+import Course from "../models/course-model.js";
+import User from "../models/user-model.js";
+import Lecture from "../models/lecture-model.js";
+const createCourse = async (req, res) => {
+  const instructorId = req.user.sub;
+  const { title, description, price, imageUrl } = req.body;
+
+  try {
+    // Ensure the instructor exists and is valid
+    const instructor = await User.findOne({
+      where: { id: instructorId, role: "instructor" },
+    });
+    if (!instructor) {
+      return res
+        .status(403)
+        .json({ message: "Instructor not found or invalid." });
+    }
+
+    // Create the course
+    const course = await Course.create({
+      title,
+      description,
+      instructorId,
+      imageUrl,
+      price,
+    });
+
+    res.status(201).json(course);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "Error creating course." });
+  }
+};
+
+const getAllCourses = async (req, res) => {
+  try {
+    // Fetch all courses along with their instructor details
+    const courses = await Course.findAll({
+      include: {
+        model: User,
+        as: "instructor",
+        attributes: ["id", "name", "email"], // Select specific fields for the instructor
+      },
+      attributes: ["id", "title", "description", "imageUrl", "createdAt"], // Select specific fields for the course
+    });
+
+    if (!courses.length) {
+      return res.status(404).json({ message: "No courses found" });
+    }
+
+    res.status(200).json(courses);
+  } catch (error) {
+    console.error("Error fetching courses:", error);
+    res.status(500).json({ error: "Failed to fetch courses" });
+  }
+};
+
+const getCoursesByInstructor = async (req, res) => {
+  try {
+    const instructorId = req.user.sub; // Extracted from the token middleware
+    if (!instructorId) {
+      return res
+        .status(404)
+        .json({ message: "Instructor ID not found in token" });
+    }
+
+    // Fetch all courses for this instructor
+    const courses = await Course.findAll({
+      where: { instructorId },
+      attributes: ["id", "title", "description", "imageUrl", "createdAt"], // Include only the required fields
```

Comment:
I think the fact that it's called `imageUrl` in the database can be a bit misleading. It should rather be a key or a reference to an image, not the entire URL. let's keep the naming for now, but please keep this in mind going forward.